provisioner/file/provisioner.go

```diff
@@ -154,7 +159,11 @@ func (p *Provisioner) ProvisionUpload(ui packer.Ui, comm packer.Communicator) er
 			return err
 		}
 
-		err = comm.Upload(p.config.Destination, f, &fi)
+		if strings.HasSuffix(dst, "/") {
+			dst = filepath.Join(dst, filepath.Base(src))
```

Comment:
Before this change in case of download direction, file copies to dst with full path.
Does example "source": "/etc/network/interfaces", "destination": "output/" copy interfaces file to output/etc/network/interfaces this is wrong and not like in scp.
So i modify dst before this to provide proper path.